includes/memcached-adapter.php

```diff
@@ -185,11 +185,31 @@ public function get_multiple( $keys, $connection_group ) {
 		$mc          = $this->get_connection( $connection_group );
 		$mapped_keys = $this->normalize_keys_with_mapping( $keys );
 
-		/** @psalm-var array<string, mixed>|false $results */
-		$results = $mc->getMulti( array_keys( $mapped_keys ) );
+		$results = [];
+		if ( count( $mapped_keys ) > 1000 ) {
+			// Sending super large multiGets to a memcached server results in
+			// extremely inflated read/response buffers which can consume a lot memory perpetually.
+			// So instead we'll chunk up and send multiple reasonably-sized requests.
```

Comment:
Noted this in Slack, but I think that instead of supporting an unbounded lookup, we should either just return `false` or only fetch values for the first `n` keys (the rest can be `false`) when the number of keys exceeds a certain amount.

Reply:
Gonna look into doing something like this, just a bit higher up where we have more context of what's happening. Specifically perhaps in WP_Query, as that's where these sorts of really high numbers are likely to come from given an unbounded or just really really large query.

If a query is returning 200k posts, then skipping the object cache lookups and term/meta cache updates makes a lot of sense. And seems more than safe to do since we know the full context of what's happening.

I'm more uncertain of doing this inside the drop-in/extension though, since we don't really know if some of the keys are more important/integral to application code.

Reply:
That said, if we set a much higher/absurd limit then I could see that making sense at this level too.

Reply:
That would only capture core-specific pathways. The same problem could manifest through direct use of `wp_cache_get_multi`.

IMO having additional safeguards in the drop-in is not a bad thing especially to protect against extremely atypical uses (since that can cause problems).

Reply:
Oh, nvm. I forgot how this plugin works :)

I guess `wp-object-cache.php` would be a better place to enforce limits.